packages/chopsticks/src/plugins/trace-transaction/index.ts

```diff
@@ -0,0 +1,168 @@
+import { Block, runTask, taskHandler } from '@acala-network/chopsticks-core'
+import { HexString } from '@polkadot/util/types'
+import { blake2AsHex } from '@polkadot/util-crypto'
+import { writeFileSync } from 'fs'
+
+import { configSchema, getYargsOptions } from '../../schema/index.js'
+import { overrideWasm } from '../../utils/override.js'
+import { setupContext } from '../../context.js'
+
+const ACALA_ETH_RPC = 'https://eth-rpc-acala.aca-api.network'
+const KARURA_ETH_RPC = 'https://eth-rpc-karura.aca-api.network'
+
+export const cli = (y: Argv) => {
+  y.command(
+    'trace-transaction <tx-hash>',
+    'EVM+ trace transaction. Only Acala and Karura are supported',
+    (yargs) =>
+      yargs.options(getYargsOptions(configSchema.shape)).positional('tx-hash', {
+        desc: 'Transaction hash',
+        type: 'string',
+      }),
+    async (argv) => {
+      const config = configSchema.parse(argv)
+      const wasm = config['wasm-override']
+      if (!wasm) {
+        throw new Error('Wasm override built with feature `tracing` is required')
+      }
+      delete config['wasm-override']
+      const context = await setupContext(config, false)
+      const txHash = argv['tx-hash']
+
+      const specName = (await context.chain.head.runtimeVersion).specName.toString()
+      let ethRpc: string | undefined
+      if (specName.includes('acala')) {
+        ethRpc = ACALA_ETH_RPC
+      } else if (specName.includes('karura')) {
+        ethRpc = KARURA_ETH_RPC
+      } else {
+        throw new Error(`Unsupported chain. Only Acala and Karura are supported`)
+      }
+      const response = await fetch(ethRpc, {
+        headers: [
+          ['Content-Type', 'application/json'],
+          ['Accept', 'application/json'],
+        ],
+        method: 'POST',
+        body: JSON.stringify({ jsonrpc: '2.0', id: 1, method: 'eth_getTransactionByHash', params: [txHash] }),
+      })
+      const data = await response.json()
+      if (data.error) {
+        throw new Error(data.error.message)
+      }
+      console.log({ transaction: data.result })
+      const { from, to, value, gas, input, blockHash } = data.result
+
+      const block = await context.chain.getBlock(blockHash)
+      if (!block) {
+        throw new Error(`Block not found ${blockHash}`)
+      }
+      const header = await block.header
+      const meta = await block.meta
+      const parent = await context.chain.getBlock(header.parentHash.toHex())
+      if (!parent) {
+        throw new Error(`Block not found ${blockHash}`)
+      }
+      await context.chain.setHead(parent)
+      await overrideWasm(context.chain, wasm)
+      const extrinsics = await block.extrinsics
+      const txIndex = extrinsics.findIndex((tx) => blake2AsHex(tx) === txHash)
+
+      const newBlock = new Block(context.chain, block.number, blockHash, parent, {
+        header,
+        extrinsics: [],
+        storage: parent.storage,
+      })
+
+      meta.registry.register({
+        Step: {
```

Comment:
the metadata from the overridden runtime should have the types?

Reply:
for some reason it's not generating types, maybe because it's not used on extrinsic function. not sure yet

Reply:
oh right. we need pjs to support metadata v15 for that to work